logistics_project/apps/malawi/warehouse/report_views/dashboard.py

```diff
@@ -68,6 +70,58 @@ def get_stockout_rates_table(self, request, child_sps, window_date):
 
         return table
 
+    def get_district_fridge_summary_table(self, request, reporting_supply_point, child_sps):
+        if request.base_level_is_facility:
+            if reporting_supply_point.type.code == SupplyPointCodes.COUNTRY:
+                districts = child_sps
+            elif reporting_supply_point.type.code == SupplyPointCodes.DISTRICT:
+                districts = [reporting_supply_point]
+            else:
+                raise BaseLevel.InvalidReportingSupplyPointException(reporting_supply_point.code)
+
+            table = {
+                "id": "district-fridge-summary",
+                "is_datatable": False,
+                "is_downloadable": False,
+                "header": [
+                    "District",
+                    "% of HFs with no gas for fridge",
+                    "% of HFs with fridge power failure",
+                    "% of HFs with fridge breakdown",
+                    "% of HFs with other fridge issue",
+                ],
+                "data": [],
+            }
+
+            for district in districts:
+                total_facilities = SupplyPoint.objects.filter(
+                    active=True,
+                    supplied_by=district
+                ).count()
+
+                malfunction_reasons = RefrigeratorMalfunction.objects.filter(
+                    supply_point__active=True,
+                    supply_point__supplied_by=district,
+                    resolved_on__isnull=True
+                ).values_list('malfunction_reason', flat=True)
+
+                reason_counts = defaultdict(lambda: 0)
+
+                for malfunction_reason in malfunction_reasons:
+                    reason_counts[malfunction_reason] += 1
```

Comment:
not blocking, but i think it would be more efficient to do the aggregation in the DB (assuming the version of django we are on supports it). Something like:
```python
malfunction_reasons.values('malfunction_reason').annotate(total=Count('malfunction_reason'))
```